src/components/RoleForm/AssignmentRestrictionsWarning.js

```diff
@@ -0,0 +1,101 @@
+import i18n from '@dhis2/d2-i18n'
+import { IconChevronDown24, IconChevronUp24, NoticeBox } from '@dhis2/ui'
+import PropTypes from 'prop-types'
+import React, { useState } from 'react'
+import { useSystemInformation } from '../../providers/index.js'
+import styles from './AssignmentRestrictionsWarning.module.css'
+
+export const AssignmentRestrictionWarning = ({
+    currentUser,
+    roleId,
+    roleAuthorities,
+    initiallyExpanded,
+}) => {
+    const [detailsExpanded, setDetailsExpanded] = useState(
+        initiallyExpanded ?? false
+    )
+
+    // check if user is able to assign this role if they are a member
+    const {
+        userRoleIds,
+        authorities: userAuthorities,
+        hasAllAuthority,
+    } = currentUser
+    const { usersCanAssignOwnUserRoles, authorityIdToNameMap } =
+        useSystemInformation()
+    const cannotAssignThisRole =
+        !hasAllAuthority &&
+        !usersCanAssignOwnUserRoles &&
+        userRoleIds.includes(roleId)
+
+    // check if role has authorities that user does not have
+    const authoritiesUserDoesNotHave = hasAllAuthority
```

Comment:
this is super small, but i really struggled with reading the const name here. Maybe missingAutoritiesForUser or something of that sort?